ethereum/ethstats/src/main/java/org/hyperledger/besu/ethstats/EthStatsService.java

```diff
@@ -274,22 +285,22 @@ private void sendFullReport() {
   private void sendPing() {
     // we store the timestamp when we sent the ping
     pingTimestamp = System.currentTimeMillis();
-    final EthStatsRequest ping =
-        new EthStatsRequest(
-            NODE_PING,
-            new PingReport(enodeURL.getNodeId().toHexString(), String.valueOf(pingTimestamp)));
-    sendMessage(webSocket, ping);
+    final PingReport pingReport =
+        ImmutablePingReport.builder()
+            .id(enodeURL.getNodeId().toHexString())
+            .currentTime(String.valueOf(pingTimestamp))
+            .build();
+    sendMessage(webSocket, new EthStatsRequest(NODE_PING, pingReport));
```

Comment:
Done. I added this in everything except in two object because having the builder allow to use this feature https://immutables.github.io/immutable.html#wrappertupple-initializers-inlined-as-alternative-setters-with-deep- immutables-detection